mesh-multi-zone/kong-mesh-multi-zone.md

```diff
@@ -0,0 +1,732 @@
+
+NOTES:
+
+In our case we create VM with ephemeral IP. Ok for testing, but **totally wrong** for production.
+Should we change our guide? (I believe so)
+
+```bash
+export GCP_PROJECT_NAME=fsi-env2
+export GCP_VM_NAME=kuma-global-cp-$RANDOM
+export GCP_ZONE=us-central1-a
+
+gcloud compute instances create $GCP_VM_NAME \
+      --project=$GCP_PROJECT_NAME \
+      --zone=$GCP_ZONE \
+      --machine-type=e2-highcpu-4 \
+      --network-interface=network-tier=PREMIUM,subnet=default \
+      --maintenance-policy=MIGRATE \
+      --provisioning-model=STANDARD \
+      --tags=$GCP_VM_NAME \
+      --create-disk=auto-delete=yes,boot=yes,device-name=$GCP_VM_NAME,image=projects/rhel-cloud/global/images/rhel-8-v20220519,mode=rw,size=20,type=projects/$GCP_PROJECT_NAME/zones/$GCP_ZONE/diskTypes/pd-balanced \
+      --no-shielded-secure-boot \
+      --shielded-vtpm \
+      --shielded-integrity-monitoring \
+      --reservation-affinity=any
+```
+
+## Create Firewall rules
+
+```bash
+gcloud compute --project=$GCP_PROJECT_NAME firewall-rules create kuma-cp-ds-$GCP_VM_NAME \
+    --description="Kuma Discovery Service port" \
+    --direction=INGRESS \
+    --priority=1000 \
+    --network=default \
+    --action=ALLOW \
+    --rules=tcp:5685 \
+    --source-ranges=0.0.0.0/0 \
+    --target-tags=$GCP_VM_NAME
+```
+
+⚠️ WARNING
+
+Open `5681` port only for testing purpose.
+
+For production need to use port `5682` (HTTPS version of the services available under 5681)
+
+```bash
+gcloud compute --project=$GCP_PROJECT_NAME firewall-rules create kuma-cp-api-$GCP_VM_NAME \
+    --description="Kuma API server (with GUI)" \
+    --direction=INGRESS \
+    --priority=1000 \
+    --network=default \
+    --action=ALLOW \
+    --rules=tcp:5681 \
+    --source-ranges=0.0.0.0/0 \
+    --target-tags=$GCP_VM_NAME
+```
+
+## Create user for Kuma
+
+Connect to Global Control Plane VM via ssh and run
+
+```bash
+sudo adduser kuma --system --create-home
+```
+
+## Configure Kuma working directory
+
+```bash
+export KUMA_WORKING_DIR=/opt/kong
+sudo mkdir -p $KUMA_WORKING_DIR
+```
+
+## Download Kuma
+
+```bash
+curl -L https://download.konghq.com/mesh-alpine/kong-mesh-1.7.0-rhel-amd64.tar.gz --output kong-mesh-1.7.0-rhel-amd64.tar.gz 
+sudo tar xvf kong-mesh-1.7.0-rhel-amd64.tar.gz -C $KUMA_WORKING_DIR
+```
+
+## Configure Kuma license
+
+Upload the license file to the virtual machine and save it to `/opt/kong/kong-mesh-1.7.0/license.json`
+
+## Configure SELinux
+
+```bash
+sudo chown kuma:kuma -R $KUMA_WORKING_DIR
+sudo restorecon -R -v $KUMA_WORKING_DIR
+```
+
+Check `kumactl` version
+
+```bash
+sudo ln -s $KUMA_WORKING_DIR/kong-mesh-1.7.0/bin/kumactl /usr/local/bin/kumactl
+kumactl version
+```
+
+Output
+
+```text
+Client: Kuma Mesh 1.7.0
+Unable to connect to control plane: Get "http://localhost:5681/": dial tcp [::1]:5681: connect: connection refused
+```
+
+## Configure systemd service
+
```

Comment:
```suggestion

The most relevant parameters when configuring the service are:

- `KUMA_ENVIRONMENT=universal` meaning it is not kubernetes
- `KUMA_MODE=global` for multizone instead of standalone
- `KMESH_LICENSE_PATH=/opt/kong/kong-mesh-1.7.0/license.json` to specify the path of the license
```